packages/visual-service/src/service.ts

```diff
@@ -356,4 +415,54 @@ export default class WdioImageComparisonService extends BaseClass {
             })
         }
     }
+
+    #getTestContext(test: Frameworks.Test | Frameworks.World): TestContext {
+        if (this.#config?.framework === 'mocha' && test) {
+            return {
+                ...this.#testContext,
+                title: (test as Frameworks.Test).title,
+                parent: (test as Frameworks.Test).parent
+            }
+        }
+        /**
+         * When using Jasmine as the framework the title/parent are not set as with mocha.
+         *
+         * `fullName` contains all describe(), and it() separated by a space.
+         * `description` contains the current it() statement.
+         *
+         * e.g.:
+         * With the following configuration
+         *
+         * describe('x', () => {
+         *   describe('y', () => {
+         *     it('z', () => {});
+         *   })
+         * })
+         *
+         * fullName will be "x y z"
+         * description will be "z"
+         *
+         */
+        if (this.#config?.framework === 'jasmine' && test) {
+            const { description: title, fullName } = test as Frameworks.Test
+
+            return {
+                ...this.#testContext,
+                title: title as string,
+                parent : fullName?.replace(` ${title}`, '') as string
+            }
+        }
+
+        if (this.#config?.framework === 'cucumber' && test) {
+            return {
+                ...this.#testContext,
+                title: (test as Frameworks.World)?.pickle?.name as string,
+                // @ts-ignore
+                parent: (test as Frameworks.World)?.gherkinDocument?.feature?.name as string,
+            }
+
+        }
+
+        return this.#testContext
```

Comment:
should be better throw an error here to make it implicit that we expect either of these frameworks?

Reply:
Thanks, will fix this